assets/js/src/segments/dynamic/dynamic_segments_filters/subscriber_subscribed_to_list.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect } from 'react';
+import { compose, get, filter } from 'lodash/fp';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+import MailPoet from 'mailpoet';
+import Select from 'common/form/select/select';
+import ReactSelect from 'common/form/react_select/react_select';
+
+import {
+  AnyValueTypes,
+  SelectOption,
+  StaticSegment,
+  WordpressRoleFormItem,
+} from '../types';
+
+export function validateSubscribedToList(formItems: WordpressRoleFormItem): boolean {
+  return (
+    (
+      (formItems.operator === AnyValueTypes.ANY)
+      || (formItems.operator === AnyValueTypes.ALL)
+      || (formItems.operator === AnyValueTypes.NONE)
+    )
+    && (
+      (Array.isArray(formItems.segments))
+      && (formItems.segments.length > 0)
+    )
+  );
+}
+
+type Props = {
+  filterIndex: number;
+}
+
+export const SubscribedToList: React.FunctionComponent<Props> = ({ filterIndex }) => {
+  const segment: WordpressRoleFormItem = useSelect(
+    (select) => select('mailpoet-dynamic-segments-form').getSegmentFilter(filterIndex),
+    [filterIndex]
+  );
+  const staticSegmentsList: StaticSegment[] = useSelect(
+    (select) => select('mailpoet-dynamic-segments-form').getStaticSegmentsList(),
+    []
+  );
+
+  const { updateSegmentFilter, updateSegmentFilterFromEvent } = useDispatch('mailpoet-dynamic-segments-form');
+
+  useEffect(() => {
+    if (
+      (
+        (segment.operator !== AnyValueTypes.ANY)
+        && (segment.operator !== AnyValueTypes.ALL)
+        && (segment.operator !== AnyValueTypes.NONE)
+      )
+    ) {
+      updateSegmentFilter({ operator: AnyValueTypes.ANY }, filterIndex);
+    }
+  }, [updateSegmentFilter, segment, filterIndex]);
+  const options = staticSegmentsList.map((currentValue) => ({
+    value: currentValue.id.toString(),
```

Comment:
There are unnecessary conversions: here you convert segment IDs to strings, later you convert them back to numbers (see my previous comments). All this can be get rid of.